pages/intro.js

```diff
@@ -0,0 +1,186 @@
+    padding-top: 17px;
+    margin: 0px;
+    background-color: ${COLOR.BACKGROUND};
+`
+
+export default function IntroPage() {
+    const [isEditingObj, setIsEditingObj] = useState({})
+    const [websiteData, setWebsiteData] = useState({})
+    // TODO need functionality to display data based on what event is currently being viewed
+
+    // TODO need to substitute the value at .doc(ID) with current website
+    const getHackathonData = async () => {
+        await db.collection('Hackathons').doc('IntroTestHackathon').get().then((doc) => {
+            setWebsiteData(doc.data().WebsiteData)
+        })
+    }
+
+    if (Object.keys(websiteData).length == 0) {
+        getHackathonData()
+    }
+
+    const handleEdit = (type) => {
+        setIsEditingObj({
+            ...isEditingObj,
+            [type]: !isEditingObj[type]
+        })
+        
+    };
+
+    const handleCancel = (type) => {
+        setIsEditingObj({
+            ...isEditingObj,
+            [type]: false
+        })
+    }
+
+    /**
+     * @param {string} type every section in the intro has a type, which is the key in the map of WebsiteData
+     * reads the value of the header/content associated with the given type and updates the object in Firebase
+     */
+    const handleSave = async (type) => {
+        const textareas = document.getElementsByClassName(type)
+        const time = new Date(new Date().getTime())
+        let updateObj = {
+            WebsiteData: {
+                ...websiteData,
+                [type]: {
+                    editor: 'GRAB USER FROM REDUX?',
+                    time, 
+                    title: websiteData[type].title
+                }
+            }
+        }
+        let header, content
+        Array.prototype.forEach.call(textareas, (textarea) => {
+            if (textarea.style.resize == 'none') {
+                // if the textarea is the header (not resizable)
+                header = textarea.value
+                updateObj.WebsiteData[type]['header'] = header
+            } else {
+                // if text area is content (resizable)
+                content = textarea.value
+                updateObj.WebsiteData[type]['content'] = content
+            }
+        })
+        // TODO need to substitute the value at .doc(ID) with current website
+        await db.collection('Hackathons').doc('IntroTestHackathon')
+            .update(updateObj)
+            .then(() => {
+                setWebsiteData({
+                    ...websiteData,
+                    [type]: {
+                        ...websiteData[type],
+                        header,
+                        content,
+                        time
+                    }
+                })
+            })
+            .catch(() => {
+                alert('An error occured while trying to update the text section')
+            })
+        setIsEditingObj({
+            ...isEditingObj,
+            [type]: false
+        })
+    }
```

Comment:
it seems weird to use await + then syntax. Maybe just use await and put it in a try/catch block to catch the error. I think that would look cleaner.